test/unit/org/apache/cassandra/cql3/conditions/ColumnConditionTest.java

```diff
@@ -321,27 +321,28 @@ public void testListCollectionBoundAppliesTo() throws InvalidRequestException
 
         // LT
         assertFalse(conditionApplies(list(ONE), LT, list(ONE)));
+        assertThrowsIRE(() -> conditionApplies(list(), LT, null), "Invalid comparison with null for operator \"<\"");
         assertFalse(conditionApplies(list(), LT, list()));
         assertFalse(conditionApplies(list(ONE), LT, list(ZERO)));
         assertTrue(conditionApplies(list(ZERO), LT, list(ONE)));
         assertFalse(conditionApplies(list(ONE, ONE), LT, list(ONE)));
         assertTrue(conditionApplies(list(ONE), LT, list(ONE, ONE)));
         assertFalse(conditionApplies(list(ONE), LT, list()));
-        assertTrue(conditionApplies(list(), LT, list(ONE)));
+        assertFalse(conditionApplies(list(), LT, list(ONE)));
```

Comment:
The behaviour is simply wrong and will need to be changed in previous versions. Comparing a null column value to list(1) should not return true otherwise it creates an inconsistency with how we handle restrictions.

Reply:
We now follow the logic of CASSANDRA-19637